sdk/cosmosdb/cosmos/src/globalEndpointManager.ts

```diff
@@ -110,7 +112,8 @@ export class GlobalEndpointManager {
   public async resolveServiceEndpoint(
     resourceType: ResourceType,
     operationType: OperationType,
-    requestContext?: RequestContext
+    requestContext?: RequestContext,
+    startIndex: number = 0
```

Comment:
It is unclear what this `startIndex` is. Pls add detailed comments explaining it's use.

Reply:
updated